NEWS.md

```diff
@@ -3,6 +3,18 @@
 
 ### Changes in v1.10.5  ( in development )
 
+#### POTENTIALLY BREAKING CHANGES
+
+1. `fread()`'s `na.strings=` argument :
+    ```
+    na.strings="NA"                           # was
+    getOption("datatable.na.strings", "NA")   # this release; i.e. no change yet
+    getOption("datatable.na.strings", "")     # future release
+    ```
+This option controls how `,,` is read in character columns. It does not affect numeric columns which read `,,` as `NA` regardless. We would like `,,`=>`NA` for consistency with numeric types, and `,"",`=>empty string to be the standard default for `fwrite/fread` character columns so that `fread(fwrite(DT))==DT` without needing any change to any parameters. `fwrite` has never written `NA` as `"NA"`, by default it already writes `,,`. The use of R's `getOption()` allows data.table users to move forward early, or restore old behaviour when the default's default is changed in future.
+
+2. `fread` now reads a column of all 0's and 1's as `logical` rather than `integer`, for convenience to avoid needing to change the type afterwards or use `colClasses`. The old behaviour can be restored with `options(datatable.logical01=FALSE)`. We felt this default change was ok to make because in all operations there should be no difference: R treats `logical` and `integer` the same. If this change does cause a problem, the option is provided to restore old behaviour while you update your code. Similarly, `fwrite` now writes `logical` columns as `0/1` by default, controlled by the same option. `0/1` is smaller and faster than `"TRUE"/"FALSE"`, which can make a significant difference to space and time the more `logical` columns there are. Further, a column of `TRUE/FALSE`s is ok, as well as `True/False`s and `true/false`s, but mixing styles (e.g. `TRUE/false`) is not and will be read as type `character`.
```

Comment:
I would be a bit more careful on the wording:

> in all operations there should be no difference: R treats `logical` and `integer` the same.

But that's not true:

```
DT = data.table(l_int = c(0, 0, 1, 0), l_log = c(FALSE, FALSE, TRUE, FALSE), i = 1:4)
DT[(l_int)]
#    l_int l_log i
# 1:     0 FALSE 1
DT[(l_log)]
#    l_int l_log i
# 1:     1  TRUE 3
```

I think (?) more accurate is that all arithmetic expecting `integer` and getting `logical` will go through as expected. Of course any function running an `is.integer` test will fail (and _vice versa_ for `is.logical` on `integer` columns).

Reply:
I had the same concern. I think as long as we add option for 1.10.6 and change its default from 1.10.8 will be fine.